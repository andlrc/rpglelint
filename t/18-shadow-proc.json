[
   {
      "column":5,
      "file":"t/18-shadow-proc.rpgle",
      "line":7,
      "message":"declaration of 'a' shadows a global declaration",
      "severity":"warning"
   },
   {
      "column":7,
      "file":"t/18-shadow-proc.rpgle",
      "line":3,
      "message":"shadowed declaration is here",
      "severity":"note"
   },
   {
      "column":5,
      "file":"t/18-shadow-proc.rpgle",
      "line":14,
      "message":"declaration of 'a' shadows a global declaration",
      "severity":"warning"
   },
   {
      "column":7,
      "file":"t/18-shadow-proc.rpgle",
      "line":3,
      "message":"shadowed declaration is here",
      "severity":"note"
   },
   {
      "column":9,
      "file":"t/18-shadow-proc.rpgle",
      "line":18,
      "message":"declaration of 'proc1' shadows a global declaration",
      "severity":"warning"
   },
   {
      "column":10,
      "file":"t/18-shadow-proc.rpgle",
      "line":5,
      "message":"shadowed declaration is here",
      "severity":"note"
   }
]
