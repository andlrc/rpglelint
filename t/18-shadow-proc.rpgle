// rpglelint: -Wshadow

dcl-s a int(10);

dcl-proc proc1;
  dcl-pi *n;
    a int(10);
    b int(10);
  end-pi;
end-proc;

dcl-proc proc2;
  dcl-pi *n;
    a int(10);
    b int(10);
  end-pi;

  dcl-s proc1 int(10);
end-proc;
