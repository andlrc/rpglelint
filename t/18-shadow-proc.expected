t/18-shadow-proc.rpgle:7:5: warning: declaration of 'a' shadows a global declaration [-Wshadow]
     a int(10);
     ^
t/18-shadow-proc.rpgle:3:7: note: shadowed declaration is here
 dcl-s a int(10);
       ^
t/18-shadow-proc.rpgle:14:5: warning: declaration of 'a' shadows a global declaration [-Wshadow]
     a int(10);
     ^
t/18-shadow-proc.rpgle:3:7: note: shadowed declaration is here
 dcl-s a int(10);
       ^
t/18-shadow-proc.rpgle:18:9: warning: declaration of 'proc1' shadows a global declaration [-Wshadow]
   dcl-s proc1 int(10);
         ^
Use of uninitialized value in printf at RPG/Linter.pm line 147.
:5:10: note: shadowed declaration is here
 dcl-proc proc1;
          ^
